moldesign/utils/classes.py

```diff
@@ -29,6 +34,67 @@ def add(self, item):
         self[key].append(item)
 
 
+class ExclusiveList(object):
+    """ Behaves like a list, but won't allow duplicate items with duplicate keys to be added.
```

Comment:
Thanks for the explanation, that makes sense to me as long as it doesn't become a bottleneck.